protocol/rpcprovider/provider_load_manager.go

```diff
@@ -40,10 +44,20 @@ func (loadManager *ProviderLoadManager) getProviderLoad() string {
 	if loadManager == nil {
 		return ""
 	}
-	loadedRateLimitThreshold := loadManager.rateLimitThreshold.Load()
-	if loadedRateLimitThreshold == 0 {
-		return "0"
+
+	return strconv.FormatUint(loadManager.activeRequestsPerSecond.Load()/loadManager.rateLimitThreshold.Load(), 10)
+}
+
+func (loadManager *ProviderLoadManager) applyProviderLoadMetadataToContextTrailer(ctx context.Context) {
+	if loadManager == nil {
```

Comment:
no need to check nil in applyProviderLoadMetadataToContextTrailer as it will just return due to getProviderLoad nil check

Reply:
Fixed in [e81afb9](https://github.com/lavanet/lava/pull/1720/commits/e81afb9004709ba9c2f23e147b30b18de21c4873)